### PR TITLE
Update make scripts and git hooks for yalc to replace yarn link

### PR DESCRIPTION
**Pull request by YuCJ**
- Do not serve service-worker on dev (the express.js will `stat` the `sw.js` file, but we won't build it when dev) 
- Add `make link-packages` and `make unlink-packages`
- Update webpack config to use yarn link without errors (To prevent webpack take the `react`, `redux`... etc moudles from `node_modules` in linked package. Ex: Taking `react-redux` from `@twreporter/redux/node_modules/react-redux`)

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -22,6 +23,25 @@ const webpackAssets =  {
...
+  'redux',
+  'react-redux',
+  'styled-components',
+  'react-router-dom',
+  'react-router',
+  'axios'
+], (prevAlias, moduleName) => ({
+  ...prevAlias,
+  [moduleName]: path.resolve(__dirname, 'node_modules', moduleName)
+}), {})
```
Comment (YuCJ) [inline webpack.config.js]: I found that this works for Webpack to build bundle. But we still may encounter some errors when server side rendering. 

```bash
make start-dev-server
 [+] start dev server by nodemon src/server.js

NODE_ENV=development RELEASE_BRANCH= RENDER_ENV=server node_modules/.bin/nodemon --delay 1800ms src/server.js --exec node_modules/.bin/babel-node
[nodemon] 1.18.10
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node_modules/.bin/babel-node src/server.js`
module.js:549
    throw err;
    ^

Error: Cannot find module 'redux'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/mac2016cjyu/Repos/twreporter-npm-packages/packages/redux/lib/store/create-store.js:8:14)
    at Module._compile (module.js:652:30)
    at loader (/Users/mac2016cjyu/Repos/twreporter-react/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/mac2016cjyu/Repos/twreporter-react/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
[nodemon] app crashed - waiting for file changes before starting...
```
I'm trying to fix this.

**Comment by YuCJ** [comment]
@babygoat found a package [`yalc`](https://github.com/whitecolor/yalc) to solve this problem.

If we want to test WIP `@twreporter/react-article-components` in `twreporter-react`, the workflow will be:

1. At `twreporter-npm-packages/packages/react-article-components`, run `yalc publish`
2. At `twreporter-react`, run `yalc add @twreporter/react-article-components`. The `yalc` will copy files in `twreporter-npm-packages/packages/react-article-components` to `twreporter-react/.yalc` and `twreporter/node_modules`. It will only take files like we publish with `npm publish`. And `yalc` will inject the linked package hash into `twreporter-react/packages.json`, so we can know we are using developing packages.
3. After tested okay, we can use `yalc remove --all` at `twreporter-react` to remove all files and injection in `twreporter-react`

**Comment by YuCJ** [comment]
> @babygoat found a package [`yalc`](https://github.com/whitecolor/yalc) to solve this problem.
> 
> If we want to test WIP `@twreporter/react-article-components` in `twreporter-react`, the workflow will be:
> 
>     1. At `twreporter-npm-packages/packages/react-article-components`, run `yalc publish`
> 
>     2. At `twreporter-react`, run `yalc add @twreporter/react-article-components`. The `yalc` will copy the files in `twreporter-npm-packages/packages/react-article-components` to `twreporter-react/.yalc` and `twreporter/node_modules`. It will only take files as if we publish with `npm publish`. And `yalc` will inject the linked package hash into `twreporter-react/packages.json`, so we can know we are using developing packages.
> 
>     3. After tested okay, we can use `yalc remove --all` at `twreporter-react` to remove all files and injection in `twreporter-react`

@nickhsine I've added some make scripts and git hooks for `yalc` (global installed package). And dropped the original commit that using webpack alias. Please help to review it. Thanks!

**Comment by nickhsine** [comment]
After thinking twice, I come up with an simple approach.
And you can compare which approach is more suitable for us.

In the packages we want to link,
such as `@twreporter/react-article-components`,
we can install alpha/beta npm package locally by the following commands,
```
$ cd ~/codes/twreporter-npm-packages/packages/react-article-components
$ npm pack
// after `npm pack`, 
// npm will generate a tarball 
// if tarball is `twreporter-react-article-components-1.0.14.tgz`

// and then
// we install the tarball in the main repo
$ cd ~/codes/twreporter-react
$ yarn  add ../twreporer-npm-packages/packages/react-article-components/twreporter-react-article-components-1.0.14.tgz
// yarn will install the tarball like usual.
```

**Comment by YuCJ** [comment]
> After thinking twice, I come up with a simple approach.
> And you can compare which approach is more suitable for us.
> 
> In the packages we want to link,
> such as `@twreporter/react-article-components`,
> we can install alpha/beta npm package locally by the following commands,
> 
> ```
> $ cd ~/codes/twreporter-npm-packages/packages/react-article-components
> $ npm pack
> // after `npm pack`, 
> // npm will generate a tarball 
> // if tarball is `twreporter-react-article-components-1.0.14.tgz`
> 
> // and then
> // we install the tarball in the main repo
> $ cd ~/codes/twreporter-react
> $ yarn  add ../twreporer-npm-packages/packages/react-article-components/twreporter-react-article-components-1.0.14.tgz
> // yarn will install the tarball like usual.
> ```

After discussion, we think that the way of how to link the packages of different repos should be decided by the developers themselves. We should not make the coupling between the project and the dev tools just for convenient (i.e. not necessary).

**Comment by YuCJ** [comment]
Add a [document](https://github.com/twreporter/between-you-and-me/blob/master/dev-notes/how-to-dev-npm-packages-comfortably.md) to this issue.

**Pull request closed without merging**